docs/dev/task_failure_recovery.md

```diff
@@ -264,4 +268,51 @@ The cluster defined restart strategy is used.
 This is helpful for streaming programs which enable checkpointing.
 By default, a fixed delay restart strategy is chosen if there is no other restart strategy defined.
 
+## Failover Strategies
+
+Flink supports different failover strategies which can be configured via the configuration parameter
+*jobmanager.execution.failover-strategy* in Flink's configuration file `flink-conf.yaml`.
+
+<table class="table table-bordered">
+  <thead>
+    <tr>
+      <th class="text-left" style="width: 50%">Failover Strategy</th>
+      <th class="text-left">Value for jobmanager.execution.failover-strategy</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+        <td>Restart all</td>
+        <td>full</td>
+    </tr>
+    <tr>
+        <td>Restart pipelined region</td>
+        <td>region</td>
+    </tr>
+  </tbody>
+</table>
+
+### Restart All Failover Strategy
+
+With this strategy, all tasks in the job will be restarted to recover from a task failure.
+
+### Restart Pipelined Region Failover Strategy
+
+With this strategy, tasks to restart depend on the regions to restart.
+
+A region is defined by this strategy as tasks that communicate via pipelined data exchanges.
+- All data exchanges in a DataStream job or Streaming Table job are pipelined.
+- All data exchanges in a Batch Table job are batched.
+- Types of data exchanges in a DataSet job is decided with the 
+  [ExecutionMode]({{ site.javadocs_baseurl }}/api/java/org/apache/flink/api/common/ExecutionMode.html) set in 
+  [ExecutionConfig]({{ site.baseurl }}/dev/execution_configuration.html).
+
+Regions to restart are decided as below:
+1. The region containing the failed task should be restarted.
```

Comment:
I'd change: *should be restarted* ->  *is restarted*

Reply:
Here we are describing what regions should be restarted.
"is restarted" have me feel that this is happening.

Not pretty sure whether my feeling is correct. 
Please correct me if I had a wrong sense of it.

Reply:
*"should"* can mean that a state is desirable or probable (see Section 1.1 or 2 in [1]).
However, restarting the region containing the failed task is a certain event. We can use will-future here, WDYT?

[1] https://www.lexico.com/en/definition/should

Reply:
I understand the concern now.
Agreed that we can use will-future.